### DISCUSSION TITLE
How to manually generate station inventory metadata for playback purposes

### DISCUSSION THREAD

**Discussion by dufangqing** [SED-EEW/SED-EEW-SeisComP-contributions#49] [Q&A] [answered]
Dear SED-EEW community,

I hope it is ok to ask a rather specific but not general question here. 

The scope of my task would be performing playback tests with the scfinder module. Since the waveform data for the test were already corrected to the physical unit of acceleration (i.e., cm/s/s), I am trying to create the corresponding station inventory from scratch. The way I created the dataless was simply setting the gain to be 100 and the unit to be m/s/s. However, I tried to use sceewenv to process the waveform data coupled with the generated inventory xml, and the results I obtained were smaller than the original miniseed as if it is scaled down by a factor, whereas I should expect very similar results theoretically.

I herein attached the dataless and miniseed I used, I would very much appreciate it if you could help me spot any possible mistakes that I have made. Thanks a lot!

[EURF.zip](https://github.com/SED-EEW/SED-EEW-SeisComP-contributions/files/14162554/EURF.zip)

**Comment by FMassin** [comment]
Just to make sure 100 m/s/s is indeed the correct gain? Is it written as overall gain in channel level or in the response level?

**Reply by FMassin** [reply]
Side note: One way to make sure metadata is correct is to use https://smp.gempa.de/.

**Reply by dufangqing** [reply]
gain was written in channel level, although I also tried to write in the response level and the same issue remains...

I first used PDCC to generate the dataless, then used dlsv2inv to get the SeisComP XML format. I imported the xml into SMP to check the gain value, sampling rates etc and they seemed correct. Do you have other insights on the possible solutions? Thanks.

**Reply by FMassin** [reply]
Could you please tell by how much the envelope values are lower than expected?

**Reply by dufangqing** [reply]
sure, in general they were about 50% smaller than expected, for some stations envelope would be even only 1/4 of the expected

**Reply by FMassin** [reply]
This doesn't look like a metadata error.... The envelope library include  high pass pre-filtering over 1/3 Hz. What is the dominant frequency of the input signal?

**Reply by dufangqing** [reply]
Dominant frequencies are ranging from 0.42 to 25 Hz in my case, mostly within 1.5 to 8 Hz more or less. For example, for the one that's the most altered after processing, the corresponding dominant frequency is 2 Hz and I wouldn’t expect much changes of the signal tbh...

**Comment by FMassin** [comment]
Dear @dufangqing 

sceewenv as a [`--dump` option](https://docs.gempa.de/sed-eew/current/apps/sceewenv.html#cmdoption-dump) that allows to redirect pre-processed data (pre-filtered and gain corrected) and final envelope data (combined and windowed) to a file in mseed format. I suggest that you use this option to inspect the problem.

More help, including output description: https://github.com/SED-EEW/SED-EEW-SeisComP-contributions/discussions/1

If the problem remain unexplained, please provide screenshots?

**Reply by dufangqing** [reply, marked as answer]
Hi Fred,

Thanks very much for your replies. The problem is solved and it was because there was not enough noise before the first arrival of my input signals, and the default 60s of the tapering window may have caused issues...

Based on my experiences, here I also summarize some necessary prep work for the input signal if one is intended to run scfinder playback (please correct me or supplement if there's any): 
1. miniseed always sorted by the end time;
2. miniseed written in 512 bytes (to avoid problems with msrtsimul);
3. guarantee at least 60s of noise before the first arrival of the input signals;
4. if the inventory XML needs to be created from scratch, standard physical units should be M, M/S and M/S**2, make sure the gain values (at the channel level) are correct, the start date of stations is before the test event, etc.
...


Cheers,
Du

**Reply by FMassin** [reply]
This is great feedback. Thanks a lot!
